chips/lowrisc/src/registers/flash_ctrl_regs.rs

```diff
@@ -0,0 +1,660 @@
+        ///   Memory property configuration for info partition in bank1,
+        (0x015c => pub(crate) bank1_info2_page_cfg: [ReadWrite<u32, BANK1_INFO2_PAGE_CFG::Register>; 2]),
+        /// HW interface info configuration rule overrides
+        (0x0164 => pub(crate) hw_info_cfg_override: ReadWrite<u32, HW_INFO_CFG_OVERRIDE::Register>),
+        /// Bank configuration registers configuration enable.
+        (0x0168 => pub(crate) bank_cfg_regwen: ReadWrite<u32, BANK_CFG_REGWEN::Register>),
+        /// Memory properties bank configuration
+        (0x016c => pub(crate) mp_bank_cfg_shadowed: [ReadWrite<u32, MP_BANK_CFG_SHADOWED::Register>; 1]),
+        /// Flash Operation Status
+        (0x0170 => pub(crate) op_status: ReadWrite<u32, OP_STATUS::Register>),
+        /// Flash Controller Status
+        (0x0174 => pub(crate) status: ReadWrite<u32, STATUS::Register>),
+        /// Current flash fsm state
+        (0x0178 => pub(crate) debug_state: ReadWrite<u32, DEBUG_STATE::Register>),
+        /// Flash error code register.
+        (0x017c => pub(crate) err_code: ReadWrite<u32, ERR_CODE::Register>),
+        /// This register tabulates standard fault status of the flash.
+        (0x0180 => pub(crate) std_fault_status: ReadWrite<u32, STD_FAULT_STATUS::Register>),
+        /// This register tabulates customized fault status of the flash.
+        (0x0184 => pub(crate) fault_status: ReadWrite<u32, FAULT_STATUS::Register>),
+        /// Synchronous error address
+        (0x0188 => pub(crate) err_addr: ReadWrite<u32, ERR_ADDR::Register>),
+        /// Total number of single bit ECC error count
+        (0x018c => pub(crate) ecc_single_err_cnt: [ReadWrite<u32, ECC_SINGLE_ERR_CNT::Register>; 1]),
+        /// Latest address of ECC single err
+        (0x0190 => pub(crate) ecc_single_err_addr: [ReadWrite<u32, ECC_SINGLE_ERR_ADDR::Register>; 2]),
+        /// Phy alert configuration
+        (0x0198 => pub(crate) phy_alert_cfg: ReadWrite<u32, PHY_ALERT_CFG::Register>),
+        /// Flash Phy Status
+        (0x019c => pub(crate) phy_status: ReadWrite<u32, PHY_STATUS::Register>),
+        /// Flash Controller Scratch
+        (0x01a0 => pub(crate) scratch: ReadWrite<u32, SCRATCH::Register>),
+        /// Programmable depth where FIFOs should generate interrupts
+        (0x01a4 => pub(crate) fifo_lvl: ReadWrite<u32, FIFO_LVL::Register>),
+        /// Reset for flash controller FIFOs
+        (0x01a8 => pub(crate) fifo_rst: ReadWrite<u32, FIFO_RST::Register>),
+        /// Current program and read fifo depth
+        (0x01ac => pub(crate) curr_fifo_lvl: ReadWrite<u32, CURR_FIFO_LVL::Register>),
+        /// Memory area: Flash program FIFO.
+        (0x01b0 => pub(crate) prog_fifo: [WriteOnly<u32>; 1]),
+        /// Memory area: Flash read FIFO.
+        (0x01b4 => pub(crate) rd_fifo: [ReadOnly<u32>; 1]),
+
+
+        (0x0000 => pub(crate) csr0_regwen: ReadWrite<u32, CSR0_REGWEN::Register>),
```

Comment:
It looks like something broke here

Reply:
@lschuermann  any idea why the tests wouldn't catch this? It's a duplicate value, that is something we should be looking for right?

Reply:
Yes, this is not added to `mod.rs`. We should probably add all of these register files there unconditionally, to make sure they're checked for syntax & type-system errors, and have their const-assertions evaluated.

Reply:
Ah, I didn't notice that. Yeah, these should all be added to the project, we don't want dead code sitting around bit rotting

Reply:
There was a bug in the register generator for peripherals that have more than one register block.  The bug doesn't matter much for C headers because you'd we expected to use the correct base address constant for the particular block.  It is, as you've pointed out, a bug for the rust register structs.

I've corrected the bug in the register generator and added all of the generated files to their respective `mod.rs` (the original intention was to add them as we convert the drivers over to using the autogenerated definitions).